pgml-extension/pgml_rust/src/api.rs

```diff
@@ -133,34 +136,39 @@ fn deploy(
         vec![(PgBuiltInOids::TEXTOID.oid(), project_name.into_datum())],
     );
     let project_id =
-        project_id.expect(format!("Project named `{}` does not exist.", project_name).as_str());
+        project_id.unwrap_or_else(|| panic!("Project named `{}` does not exist.", project_name));
     let task = Task::from_str(&task.unwrap()).unwrap();
 
     let mut sql = "SELECT models.id, models.algorithm::TEXT FROM pgml_rust.models JOIN pgml_rust.projects ON projects.id = models.project_id".to_string();
     let mut predicate = "\nWHERE projects.name = $1".to_string();
-    match algorithm {
-        Some(algorithm) => {
-            predicate += &format!(
-                "\nAND algorithm::TEXT = '{}'",
-                algorithm.to_string().as_str()
-            )
-        }
-        _ => (),
+    if let Some(algorithm) = algorithm {
+        let _ = write!(
+            predicate,
+            "\nAND algorithm::TEXT = '{}'",
+            algorithm.to_string().as_str()
+        );
     }
     match strategy {
         Strategy::best_score => match task {
             Task::regression => {
-                sql += &format!("{predicate}\nORDER BY models.metrics->>'r2' DESC NULLS LAST");
+                let _ = write!(
```

Comment:
What come on clippy, this seems totally fine.

Reply:
format! does an extra allocation compared to write... 🤷‍♂️